src/middlewares/password-hasher.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const errors = require('@feathersjs/errors');
+const hashFactory = require('../helpers/hash/hash-factory')();
+
+module.exports = function createPasswordHasher(requirePassword = true) {
+  if (requirePassword) {
+    return async function hashPassword(request, response, next) {

```

Comment:
Good job on creating a function that is returning an other function, but it is not a good Idea usually to send a boolean parameter to a function this in most cases implies that your function is doing two different things (violation of SRP). So here it will be better to have 2 separate functions just export them 
```
module.exports = {
   hashPasswordOptional,
   hashPassword
}
```

Reply:
ok